client/engine/store/durablestore.go

```diff
@@ -34,36 +34,57 @@ type DurableStore struct {
 
 // NewDurableStore creates a new DurableStore that uses the given folder to store its data
 // It will create the folder if it does not exist
-func NewDurableStore(key []byte, folder string, config buntdb.Config) Store {
+func NewDurableStore(key []byte, folder string, config buntdb.Config) (Store, error) {
```

Comment:
Since you've made the change already, I think this is fine. But otherwise probably a case of being ok to panic in a constructor?

 It's hard to think of the calling function handling the error and the program being able to continue. I suppose in theory it could fail over to a memstore if the durable store construction failed...